guides/keras_cv/stl10_with_simclr.py

```diff
@@ -0,0 +1,205 @@
+ `include_top=False` and enabling pooling on the output. In this example, we use
+a ResNet50V2 from KerasCV with `pooling="avg"`
+"""
+encoder = models.ResNet50V2(
+    include_rescaling=True,
+    include_top=False,
+    input_shape=IMAGE_SIZE + (3,),
+    pooling="avg",
+)
+
+"""
+### Creating our Trainer
+
+Next, we can create a contrastive trainer. KerasCV offers a flexible `ContrastiveTrainer`
+API in addition to a `SimCLRTrainer`, which extends `ContrastiveTrainer` and
+provides the default structure of SimCLR. KerasCV also implements a `SimCLRAugmenter`
+which performs the augmentation steps which showed the best results in the SimCLR
+paper.
+
+We use a `SimCLRTrainer` with a `SimCLRAugmenter` in this example, which gives us
+the power of SimCLR with minimal manual configuration. Users who wish to customize
+their contrastive augmenters and projectors further can use a `ContrastiveTrainer`
+directly.
+"""
+trainer = training.SimCLRTrainer(
+    encoder=encoder,
+    augmenter=training.SimCLRAugmenter(
+        value_range=(0, 255), target_size=IMAGE_SIZE
+    ),
+)
+
+"""
+### Training Configuration
+
+Next, we set up training configuration including optimizer selection, loss, and
+callbacks.
+
+For our loss function we use the loss function used in the SimCLR paper,
+"Normalized Temperature-scaled Cross Entropy loss". KerasCV packages this as `SimCLRLoss`.
+
+Note that we perform gradient clipping in our optimizer. In our experiments, we found
+gradient clipping reduced gradient explosion when performing training with a probe.
+
+In this guide, as we're training on unlabelled images, no probe is used.
+"""
+
+trainer.compile(
+    encoder_optimizer=optimizers.SGD(
+        learning_rate=0.05, momentum=0.9, global_clipnorm=10
+    ),
+    encoder_loss=losses.SimCLRLoss(temperature=0.5),
+)
+
+training_callbacks = [
+    callbacks.EarlyStopping(monitor="loss", patience=5, min_delta=0.1),
+    callbacks.BackupAndRestore("backup"),
+    callbacks.TensorBoard(log_dir="logs"),
+]
+
+"""
+### Fitting our Model
+
+Now, we can simply call `trainer.fit` to perform the contrastive learning process
+using our unlabelled images.
+"""
+trainer.fit(
+    unlabelled_images.take(100),
+    epochs=CONTRASTIVE_EPOCHS,
+    callbacks=training_callbacks,
+)
+
+
+"""
+## Fine-Tuning
+
+Now that we've performed contrastive feature learning, we can train a classifier
+using our pre-trained encoder as input. To do so, we freeze the pre-trained encoder
+and add a Dense layer to it to perform classification.
+"""
+
+encoder.trainable = False
+classification_model = keras.Sequential(
+    [encoder, layers.Dense(CLASSES, activation="softmax")]
+)
+
+classification_model.compile(
+    optimizer="adam", loss=keras.losses.CategoricalCrossentropy()
+)
+
+"""
+Then, we simply call `fit` on our new classification model using our labelled images.
+"""
+
+classification_model.fit(train_labelled_images)
+
+classification_model.evaluate(test_labelled_images)
```

Comment:
Please add a conclusion sentence / paragraph (e.g. providing the metrics you'd expect after a full run, and listing what you want the reader to take away)